Makefile

```diff
@@ -0,0 +1,39 @@
+# Makefile for running typical developer workflow actions.
+# To run actions in a subdirectory of the repo:
+#   make lint build dir=translate/snippets
+
+INTERFACE_ACTIONS="build test lint"
+
+.ONESHELL: #ease subdirectory work by using the same subshell for all commands
+.-PHONY: *
+
+# Default to current dir if not specified.
+dir ?= $(shell pwd)
+
+export GOOGLE_CLOUD_PROJECT = ${GOOGLE_SAMPLES_PROJECT}
+
+build:
+	cd ${dir}
+	mvn compile
+
+test: check-env build
+	cd ${dir}
+	mvn --quiet --batch-mode --fail-at-end clean verify \
+    -Dfile.encoding="UTF-8" \
+    -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+    -Dmaven.test.redirectTestOutputToFile=true \
+    -Dbigtable.projectID="${GOOGLE_CLOUD_PROJECT}" \
+    -Dbigtable.instanceID=instance
+
+lint:
+	cd ${dir}
+	mvn -P lint checkstyle:check
+
+check-env:
```

Comment:
Suggesting to add a check for "mvn". If not, provide a link to the installation page?

Reply:
This is a nice enhancement. Is there a way to do this that will work also work on Windows?
(I'm not very familiar with the Windows dev process - do you know anyone that could help us verify that these work on windows?)

Reply:
Parking this for addition in the future PRs.